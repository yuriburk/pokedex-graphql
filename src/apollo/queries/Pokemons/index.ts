import { gql } from '@apollo/client';

export const GET_POKEMONS = gql`
  query GetPokemons($count: Int!) {
    pokemons(first: $count) {
      id
      image
      name
    }
  }
`;

export const GET_POKEMONS_UPDATED = gql`
  query GetPokemonsUpdated {
    pokemonsUpdated @client
  }
`;
