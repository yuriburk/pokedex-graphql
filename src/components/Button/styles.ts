import styled from 'styled-components';

export const Container = styled.button`
  background: #78c850;
  height: 54px;
  border-radius: 10px;
  border: 0;
  padding: 0 14px;
  width: 100%;
  color: black;
  font-weight: bold;
  margin-top: 16px;
  transition: background-color 0.2 ease 0.2;

  &:hover {
    background: '#78c850';
  }
`;
