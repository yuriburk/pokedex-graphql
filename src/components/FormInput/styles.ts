import styled, { css } from 'styled-components';

import Tooltip from '../Tooltip';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
`;

export const Label = styled.label`
  font-weight: bold;
  margin-bottom: 12px;
`;

interface IInputContainerProps {
  isFocused: boolean;
  isFilled: boolean;
  isErrored: boolean;
}

export const InputContainer = styled.div<IInputContainerProps>`
  background: #232129;
  border-radius: 10px;
  padding: 14px;
  border: 2px solid #232129;
  color: ${(props) =>
    props.isFocused || props.isFilled ? '#ff9000' : '#666360'};
  ${(props) =>
    props.isErrored &&
    css`
      border-color: #c53030;
    `}
  ${(props) =>
    props.isFocused &&
    css`
      border-color: #ff9000;
    `}

  display: flex;
  align-items: center;

  & + div {
    margin-top: 12px;
  }

  input {
    flex: 1;
    background: transparent;
    border: 0;
    color: #f4ede8;
    width: 100%;

    &::placeholder {
      color: #666360;
    }

    &:-webkit-autofill,
    &:-webkit-autofill:hover,
    &:-webkit-autofill:focus,
    &:-webkit-autofill:active {
      -webkit-box-shadow: 0 0 0 50px #232129 inset;
      -webkit-text-fill-color: #f4ede8;
    }
  }

  svg {
    margin-right: 16px;
  }
`;

export const Error = styled(Tooltip)`
  height: 20px;
  margin-left: 16px;

  svg {
    margin: 0;
  }

  span {
    background: #c53030;
    color: #fff;

    &::before {
      border-color: #c53030 transparent;
    }
  }
`;
