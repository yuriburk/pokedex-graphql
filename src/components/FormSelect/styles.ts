import { CSSProperties } from 'react';
import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;

  margin-bottom: 8px;
`;

export const Label = styled.label`
  font-weight: bold;
  margin-bottom: 2px;
`;

export const customStyles = {
  container: (base: CSSProperties): CSSProperties => ({
    ...base,
    flex: 1,
  }),
  control: (base: CSSProperties): Record<string, unknown> => ({
    ...base,
    backgroundColor: '#232129',
    borderColor: '#232129',
    padding: '6px',
    boxShadow: 'none',
    borderRadius: '10px',
    '&:hover': {
      borderColor: '#ff9000',
    },
  }),
  multiValue: (base: CSSProperties): CSSProperties => ({
    ...base,
    backgroundColor: '#1d1c24',
  }),
  multiValueLabel: (base: CSSProperties): CSSProperties => ({
    ...base,
    color: '#fff',
  }),
  menuList: (base: CSSProperties): CSSProperties => ({
    ...base,
    color: '#fff',
    backgroundColor: '#232129',
  }),
  option: (base: CSSProperties): Record<string, unknown> => ({
    ...base,
    backgroundColor: '#232129',
    '&:hover': {
      backgroundColor: '#1d1c24',
    },
  }),
  singleValue: (base: CSSProperties): Record<string, unknown> => ({
    ...base,
    color: '#fff',
  }),
};
