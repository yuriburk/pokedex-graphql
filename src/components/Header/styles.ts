import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 18px;

  svg {
    cursor: pointer;
  }
`;
