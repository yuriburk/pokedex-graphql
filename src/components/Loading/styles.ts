import styled from 'styled-components';

export const LoadingContainer = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
`;
