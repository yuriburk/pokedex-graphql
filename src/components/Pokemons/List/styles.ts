import styled, { css } from 'styled-components';
import { IPokemonTypes } from 'interfaces';

export const Container = styled.div``;

export const List = styled.ul`
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;

  list-style: none;
`;

export const ListItem = styled.li`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  flex: 1;

  cursor: pointer;
  min-height: 289px;
  padding: 14px 30px;
  border-radius: 8px;
  margin: 4px;
  background-color: ${(props) => props.theme.colors.secundaryDark};

  transition: transform ease 0.5s;

  &:hover {
    transform: translateY(-6px);
  }
`;

export const PokemonInfo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const Title = styled.h2`
  color: ${(props) => props.theme.colors.primaryWhite};
  margin-top: 8px;
  word-break: break-word;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
`;

export const TextInfo = styled.p`
  color: white;
`;

export const SpecialContainer = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;

  margin-bottom: 22px;
`;

interface ISpecialInfoContainerProps {
  pokemonType: IPokemonTypes;
}

export const SpecialInfoContainer = styled.div<ISpecialInfoContainerProps>`
  padding: 4px 8px;
  margin: 4px 4px 0 0;
  border-radius: 12px;
  ${(props) => props.theme.typesColors[props.pokemonType]};
`;

export const SpecialInfo = styled.p`
  color: ${(props) => props.theme.colors.primaryWhite};
`;

export const Image = styled.img`
  width: 124px;
  height: 124px;
  border-radius: 8px;
`;

export const ButtonContainer = styled.div`
  display: flex;
  justify-content: flex-end;
`;
