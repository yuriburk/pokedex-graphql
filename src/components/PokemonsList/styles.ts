import styled from 'styled-components';

export const Container = styled.ul`
  display: flex;
  flex-wrap: wrap;

  list-style: none;
`;

export const Item = styled.li`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  flex: 1;

  padding: 15px;
  border-radius: 5px;
  margin: 5px;

  background-color: white;
`;

export const Image = styled.img`
  width: 125px;
  height: 125px;
`;

export const Title = styled.h2`
  color: black;

  margin-top: 15px;
`;
