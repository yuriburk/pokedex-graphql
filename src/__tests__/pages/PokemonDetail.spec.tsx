import React from 'react';
import { render, cleanup, fireEvent, waitFor } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';

import { GET_POKEMONS } from 'operations/queries/Pokemons/server';
import PokemonDetail from 'pages/PokemonDetail';
import { theme } from 'styles/theme';
import { IPokemon } from 'interfaces';
import { pokemonsStore } from 'operations';

const mockTestId = 'test-id';
const mockEvolutionTestId = 'test-id2';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: mockHistoryPush,
  }),
  useParams: jest.fn(() => ({ id: mockTestId })),
  Link: ({ children }: { children: React.ReactNode }) => children,
}));

const pikachu: IPokemon = {
  id: mockTestId,
  name: 'Pikachu',
  image: 'image',
  number: '001',
  weaknesses: ['Rock'],
  classification: 'Mouse Pokémon',
  maxCP: 100,
  maxHP: 400,
  resistant: ['Water'],
  attacks: {
    special: [
      {
        name: 'Thundershock',
        damage: 90,
        type: 'Electric',
      },
    ],
  },
  weight: {
    minimum: '0.1kg',
    maximum: '4kg',
  },
  height: {
    minimum: '0.1m',
    maximum: '0.5m',
  },
  evolutions: [
    {
      id: mockEvolutionTestId,
    },
  ],
};
const raichu: IPokemon = {
  id: mockEvolutionTestId,
  name: 'Raichu',
  image: 'image',
  number: '002',
  weaknesses: ['Fire'],
  classification: 'Mouse Pokémon',
  maxCP: 100,
  maxHP: 400,
  resistant: ['Water'],
  attacks: {
    special: [
      {
        name: 'Tackle',
        damage: 40,
        type: 'Normal',
      },
    ],
  },
  weight: {
    minimum: '0.1kg',
    maximum: '4kg',
  },
  height: {
    minimum: '0.1m',
    maximum: '0.5m',
  },
};
const pokemons: IPokemon[] = [pikachu, raichu];

const PokemonListWrapper = () => (
  <MockedProvider addTypename={false}>
    <ThemeProvider theme={theme}>
      <PokemonDetail />
    </ThemeProvider>
  </MockedProvider>
);

describe('PokemonDetail Component', () => {
  beforeAll(() => pokemonsStore(pokemons));

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it('should be able to take a snapshot', () => {
    const { container } = render(<PokemonListWrapper />);

    expect(container).toMatchSnapshot();
  });

  it('should be able to update pokémon', async () => {
    const { getByDisplayValue, getByText } = render(<PokemonListWrapper />);

    const nameInput = getByDisplayValue(pikachu.name);
    const saveButton = getByText('Salvar');

    fireEvent.change(nameInput, { target: { value: 'Novo Pikachu' } });

    fireEvent.click(saveButton);

    await waitFor(() => expect(mockHistoryPush).toHaveBeenCalledWith('/'));
  });

  it('should be able to validate pokémon fields on update', async () => {
    pikachu.attacks.special[0].type = undefined;

    pokemonsStore([pikachu, raichu]);
    const { getByDisplayValue, getByTestId, getByText } = render(
      <PokemonListWrapper />,
    );

    const nameInput = getByDisplayValue(pikachu.name);
    const nameInputContainer = getByTestId('input-container-name');
    const saveButton = getByText('Salvar');

    fireEvent.change(nameInput, { target: { value: '' } });

    fireEvent.click(saveButton);

    await waitFor(() => {
      expect(mockHistoryPush).not.toHaveBeenCalled();
      expect(nameInputContainer).toHaveStyle('border-color: #c53030;');
    });
  });
});
