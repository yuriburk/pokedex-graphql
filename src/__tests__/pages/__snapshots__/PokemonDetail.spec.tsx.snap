// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetail Component should be able to take a snapshot 1`] = `
<div>
  <div
    class="sc-fzpjYC gpmpYc"
  >
    <div
      class="sc-fzoXzr bweXQK"
    >
      <h1>
        Details
      </h1>
      <svg
        fill="none"
        height="42"
        size="42"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="19"
          x2="5"
          y1="12"
          y2="12"
        />
        <polyline
          points="12 19 5 12 12 5"
        />
      </svg>
    </div>
    <form
      class="sc-fznxsB grqhhL"
    >
      <div
        class="sc-fznJRM cHmYoj"
      >
        <img
          alt="Pikachu"
          class="sc-fznWqX jtnVd"
          src="image"
        />
        <h2>
          Pikachu
        </h2>
        <p
          class="sc-fzoiQi bdygDM"
        >
          #
          001
        </p>
      </div>
      <div
        class="sc-fzqARJ bHlPUo"
      >
        <fieldset
          class="sc-fzqNqU inCzMk"
        >
          <div
            class="sc-fzoyTs kFtWTx"
          >
            <div
              class="sc-AxirZ jFeNgR"
            >
              <label
                class="sc-AxiKw cJozpM"
                for="name"
              >
                Name
              </label>
              <div
                class="sc-AxhCb LJpH"
                data-testid="input-container-name"
              >
                <input
                  name="name"
                  value="Pikachu"
                />
              </div>
            </div>
            <div
              class="sc-AxirZ jFeNgR"
            >
              <label
                class="sc-AxiKw cJozpM"
                for="number"
              >
                Number
              </label>
              <div
                class="sc-AxhCb LJpH"
                data-testid="input-container-number"
              >
                <input
                  name="number"
                  value="001"
                />
              </div>
            </div>
            <div
              class="sc-AxirZ jFeNgR"
            >
              <label
                class="sc-AxiKw cJozpM"
                for="classification"
              >
                Classification
              </label>
              <div
                class="sc-AxhCb LJpH"
                data-testid="input-container-classification"
              >
                <input
                  name="classification"
                  value="Mouse PokÃ©mon"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-fzoyTs kFtWTx"
          >
            <div
              class="sc-AxirZ jFeNgR"
            >
              <label
                class="sc-AxiKw cJozpM"
                for="image"
              >
                Image
              </label>
              <div
                class="sc-AxhCb LJpH"
                data-testid="input-container-image"
              >
                <input
                  name="image"
                  value="image"
                />
              </div>
            </div>
            <div
              class="sc-AxirZ jFeNgR"
            >
              <label
                class="sc-AxiKw cJozpM"
                for="maxCP"
              >
                Max CP
              </label>
              <div
                class="sc-AxhCb LJpH"
                data-testid="input-container-maxCP"
              >
                <input
                  name="maxCP"
                  value="100"
                />
              </div>
            </div>
            <div
              class="sc-AxirZ jFeNgR"
            >
              <label
                class="sc-AxiKw cJozpM"
                for="maxHP"
              >
                Max HP
              </label>
              <div
                class="sc-AxhCb LJpH"
                data-testid="input-container-maxHP"
              >
                <input
                  name="maxHP"
                  value="400"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-fzoyTs kFtWTx"
          >
            <div
              class="sc-AxirZ jFeNgR"
            >
              <label
                class="sc-AxiKw cJozpM"
                for="weight.minimum"
              >
                Min Weight
              </label>
              <div
                class="sc-AxhCb LJpH"
                data-testid="input-container-weight.minimum"
              >
                <input
                  name="weight.minimum"
                  value="0.1kg"
                />
              </div>
            </div>
            <div
              class="sc-AxirZ jFeNgR"
            >
              <label
                class="sc-AxiKw cJozpM"
                for="weight.maximum"
              >
                Max Weight
              </label>
              <div
                class="sc-AxhCb LJpH"
                data-testid="input-container-weight.maximum"
              >
                <input
                  name="weight.maximum"
                  value="4kg"
                />
              </div>
            </div>
            <div
              class="sc-AxirZ jFeNgR"
            >
              <label
                class="sc-AxiKw cJozpM"
                for="height.minimum"
              >
                Min Height
              </label>
              <div
                class="sc-AxhCb LJpH"
                data-testid="input-container-height.minimum"
              >
                <input
                  name="height.minimum"
                  value="0.1m"
                />
              </div>
            </div>
            <div
              class="sc-AxirZ jFeNgR"
            >
              <label
                class="sc-AxiKw cJozpM"
                for="height.maximum"
              >
                Max Height
              </label>
              <div
                class="sc-AxhCb LJpH"
                data-testid="input-container-height.maximum"
              >
                <input
                  name="height.maximum"
                  value="0.5m"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-AxgMl fzotaY"
          >
            <label
              class="sc-AxheI HPnrV"
              for="resistant"
            >
              Resistant
            </label>
            <div
              class=" css-1akvy1b-container"
            >
              <div
                class="react-select__control css-2rzzny-control"
              >
                <div
                  class="react-select__value-container react-select__value-container--is-multi react-select__value-container--has-value css-g1d714-ValueContainer"
                >
                  <div
                    class="css-nmecjr-multiValue react-select__multi-value"
                  >
                    <div
                      class="css-11547y4 react-select__multi-value__label"
                    >
                      Water
                    </div>
                    <div
                      class="css-xb97g8 react-select__multi-value__remove"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-6q0nyr-Svg"
                        focusable="false"
                        height="14"
                        viewBox="0 0 20 20"
                        width="14"
                      >
                        <path
                          d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-b8ldur-Input"
                  >
                    <div
                      class="react-select__input"
                      style="display: inline-block;"
                    >
                      <input
                        aria-autocomplete="list"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        id="resistant"
                        spellcheck="false"
                        style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                >
                  <div
                    aria-hidden="true"
                    class="react-select__indicator react-select__clear-indicator css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                      />
                    </svg>
                  </div>
                  <span
                    class="react-select__indicator-separator css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-AxgMl fzotaY"
          >
            <label
              class="sc-AxheI HPnrV"
              for="weaknesses"
            >
              Weakness
            </label>
            <div
              class=" css-1akvy1b-container"
            >
              <div
                class="react-select__control css-2rzzny-control"
              >
                <div
                  class="react-select__value-container react-select__value-container--is-multi css-g1d714-ValueContainer"
                >
                  <div
                    class="react-select__placeholder css-1wa3eu0-placeholder"
                  >
                    Select weakness types
                  </div>
                  <div
                    class="css-b8ldur-Input"
                  >
                    <div
                      class="react-select__input"
                      style="display: inline-block;"
                    >
                      <input
                        aria-autocomplete="list"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        id="weaknesses"
                        spellcheck="false"
                        style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class="react-select__indicator-separator css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <h4>
            Specials
          </h4>
          <div
            class="sc-fzoyTs kFtWTx"
          >
            <div
              class="sc-fzoNJl zAzro"
            >
              <div
                class="sc-AxirZ jFeNgR"
              >
                <label
                  class="sc-AxiKw cJozpM"
                  for="attacks.special[0].name"
                >
                  Name
                </label>
                <div
                  class="sc-AxhCb LJpH"
                  data-testid="input-container-attacks.special[0].name"
                >
                  <input
                    name="attacks.special[0].name"
                    value="Thundershock"
                  />
                </div>
              </div>
              <div
                class="sc-AxirZ jFeNgR"
              >
                <label
                  class="sc-AxiKw cJozpM"
                  for="attacks.special[0].damage"
                >
                  Damage
                </label>
                <div
                  class="sc-AxhCb LJpH"
                  data-testid="input-container-attacks.special[0].damage"
                >
                  <input
                    name="attacks.special[0].damage"
                    value="90"
                  />
                </div>
              </div>
              <div
                class="sc-AxgMl fzotaY"
              >
                <label
                  class="sc-AxheI HPnrV"
                  for="attacks.special[0].type"
                >
                  Type
                </label>
                <div
                  class=" css-1akvy1b-container"
                >
                  <div
                    class="react-select__control css-2rzzny-control"
                  >
                    <div
                      class="react-select__value-container react-select__value-container--has-value css-g1d714-ValueContainer"
                    >
                      <div
                        class="react-select__single-value css-1opnhvy-singleValue"
                      >
                        Electric
                      </div>
                      <div
                        class="css-b8ldur-Input"
                      >
                        <div
                          class="react-select__input"
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="attacks.special[0].type"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="react-select__indicator-separator css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-6q0nyr-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-fzoXWK ggdMuR"
          >
            <h4>
              Evolutions
            </h4>
            <div
              class="sc-Axmtr hBVTto"
            >
              <ul
                class="sc-AxmLO dvUWxE"
              >
                <li
                  class="sc-fzozJi bEBSWk"
                  data-testid="list-item-Raichu"
                  style="background-color: rgb(35, 33, 41);"
                >
                  <div
                    class="sc-fzoLsD bcnFzE"
                  >
                    <img
                      alt="Raichu"
                      class="sc-fzokOt cZKTvg"
                      src="image"
                    />
                    <h2
                      class="sc-fzpans joqlYm"
                    >
                      Raichu
                    </h2>
                    <p
                      class="sc-fzplWN lcaWxL"
                    >
                      #
                      002
                    </p>
                  </div>
                  <div
                    class="sc-fznyAO iDRvJd"
                  >
                    <div
                      class="sc-fznKkj ipeEIj"
                    >
                      <p
                        class="sc-fznZeY LRfVQ"
                      >
                        Normal
                      </p>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </fieldset>
        <button
          class="sc-fzoLag isDFBH"
          type="submit"
        >
          Salvar
        </button>
      </div>
    </form>
  </div>
</div>
`;
